pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,514 @@
+	clientDeploymentName        = "test-client"
+	echoSameNodeDeploymentName  = "echo-same-node"
+	echoOtherNodeDeploymentName = "echo-other-node"
+	kindEchoName                = "echo"
+	kindClientName              = "client"
+	agentDaemonSetName          = "antrea-agent"
+)
+
+func (k *k8sConnectivityParams) Run(ctx context.Context) error {
+	c, err := k.initClients(ctx)
+	if err != nil {
+		return err
+	}
+	k.clients = c
+
+	err = k.deploy(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err := k.validateDeployment(ctx); err != nil {
+		return err
+	}
+	k.validatePodToPod(ctx)
+	k.validatePodInternetConnectivity(ctx)
+	k.Log("Test finished")
+	k.Log("Deleting deployments")
+	if err := k.deleteDeployments(ctx, k8sClient); err != nil {
+		k.Log("Deployments deletion failed")
+	} else {
+		k.Log("Deployments deletion successful")
+	}
+	return nil
+}
+
+func agnhostConnectCommand(target string) []string {
+	return []string{"/agnhost", "connect", target, "--timeout=5s"}
+}
+
+func newService(name string, selector map[string]string, portName string, port int) *corev1.Service {
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: corev1.ServiceSpec{
+			Type: corev1.ServiceTypeClusterIP,
+			Ports: []corev1.ServicePort{
+				{Name: name, Port: int32(port)},
+			},
+			Selector: selector,
+		},
+	}
+}
+
+type deploymentParameters struct {
+	Name        string
+	Kind        string
+	Image       string
+	Replicas    int
+	Port        int
+	Command     []string
+	Affinity    *corev1.Affinity
+	Tolerations []corev1.Toleration
+	Labels      map[string]string
+}
+
+func newDeployment(p deploymentParameters) *appsv1.Deployment {
+	if p.Replicas == 0 {
+		p.Replicas = 1
+	}
+	replicas32 := int32(p.Replicas)
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: p.Name,
+			Labels: map[string]string{
+				"name": p.Name,
+				"kind": p.Kind,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: p.Name,
+					Labels: map[string]string{
+						"name": p.Name,
+						"kind": p.Kind,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name: p.Name,
+							Env: []corev1.EnvVar{
+								{Name: "PORT", Value: fmt.Sprintf("%d", p.Port)},
+							},
+							Ports: []corev1.ContainerPort{
+								{ContainerPort: int32(p.Port)},
+							},
+							Image:           p.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         p.Command,
+						},
+					},
+					Affinity: p.Affinity,
+				},
+			},
+			Replicas: &replicas32,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": p.Name,
+					"kind": p.Kind,
+				},
+			},
+		},
+	}
+}
+
+type k8sConnectivityParams struct {
+	clients         *deploymentClients
+	client          k8sClientOperations
+	params          Parameters
+	antreaNamespace string
```

Comment:
There are two "antreaNamespace". I think you could move members of `Parameters` here.

Reply:
done